tests/app/celery/test_scheduled_tasks.py

```diff
@@ -13,19 +15,63 @@
     delete_invitations,
     timeout_notifications,
     run_scheduled_jobs,
-    send_daily_performance_platform_stats
+    send_daily_performance_platform_stats,
+    switch_current_sms_provider_on_slow_delivery
 )
 from app.clients.performance_platform.performance_platform_client import PerformancePlatformClient
 from app.dao.jobs_dao import dao_get_job_by_id
+from app.dao.provider_details_dao import (
+    dao_update_provider_details,
+    get_current_provider
+)
 from app.utils import get_london_midnight_in_utc
 from tests.app.conftest import (
     sample_notification as create_sample_notification,
     sample_job as create_sample_job,
-    sample_notification_history as create_notification_history
+    sample_notification_history as create_notification_history,
+    sample_service as create_sample_service,
+    sample_template as create_sample_template
 )
+from tests.conftest import set_config_values
 from unittest.mock import call, patch, PropertyMock
 
 
+def _create_slow_delivery_notification(notify_db, notify_db_session, provider='mmg'):
+    now = datetime.utcnow()
+    five_minutes_from_now = now + timedelta(minutes=5)
+
+    service = create_sample_service(
```

Comment:
if you use the `tests/db.py` functions you won't need `notify_db` or `notify_db_session` in these function calls :)